model/config.go

```diff
@@ -203,6 +203,7 @@ type ServiceSettings struct {
 	EnableUserTypingMessages                 *bool
 	EnableChannelViewedMessages              *bool
 	EnableUserStatuses                       *bool
+	ExperimentalEnableAuthenticationTransfer *bool
```

Comment:
Why is this Experimental? We have had this ability for a while. This is just a new setting to disable it.

Reply:
@crspeller We're likely to remove this config setting in the future, or replace it with something else when we add guest accounts. We're adding it for now to support Shift's use case.

Reply:
How would guest accounts effect this setting? If we are just going to remove it we shouldn't add it in the first place. 

If we really want to call the setting excremental. It should be ExperimentalDisableAuthenticationTransfer. Otherwise it sounds like the feature is experimental.

Reply:
@crspeller Appreciate the feedback.

> How would guest accounts effect this setting?

There are some use cases where staff accounts might auth via SAML, guest accounts via email/password, and you want to disable auth transfer for one of those groups. It may or may not impact this setting, but until we've designed the feature we won't know for sure.

Also I want to rephrase my words for `We're likely to remove...`: I should have said there is a chance we replace/remove this setting in the future. For now we're taking this setting to cover Shift's use case. Since we may not keep it, we want to use the term `Experimental` in the config name to make that possibility clear. We've done so before, e.g. `ExperimentalTownSquareIsReadOnly `.

> If we really want to call the setting experimental. It should be ExperimentalDisableAuthenticationTransfer. Otherwise it sounds like the feature is experimental.

Fair enough. The concern I have is that every other boolean type setting enables the feature. This would do the opposite.

Would `ExperimentalAllowAuthenticationTransfer` work instead? We've used `Allow` for other settings and I think serves the same purpose as `Disable`.

Reply:
Alright. I guess it's fine as is.